CONTRIBUTING.md

```diff
@@ -3,28 +3,48 @@
 All information related to ECS is versioned in the [elastic/ecs](https://github.com/elastic/ecs) repository. All changes to ECS
 happen through Pull Requests submitted through Git.
 
+ECS is an open source project and we love to receive contributions from our community - you!
 
-## Requirements
+## Table of Contents
+
+- [How to Contribute](#how-to-contribute)
+  - [Dev Tools](#dev-tools)
+  - [Submitting Changes](#submitting-changes)
+- [Git and Github Guidelines](#git-and-github-guidelines)
+  - [Forking](#forking)
+  - [Commits and Merging](#commits-and-merging)
+  - [Issues](#issues)
+- [Documentation](#documentation)
+- [Schema Files](#schema-files)
+- [Additional Resources](#additional-resources)
+
+## How to Contribute
+
+### Dev Tools
 
 You need these tools to contribute to ECS:
 
 * [Git](https://git-scm.com/)
 * [Python 3.6+](https://www.python.org/)
 * [Go 1.13](https://golang.org/)
 
-## Steps to contribute
-
-Here are the steps for contributing to ECS.
+### Submitting Changes
 
+* Sign the [Contributor License Agreement](http://www.elastic.co/contributor-agreement/).
 * Set up your git environment.
   - Create [your own fork](https://help.github.com/articles/fork-a-repo/) of the ECS repo.
   - Clone your fork to your machine.
 * Create a local branch to hold your changes.
   - Run `git checkout -b branch-name`, where `branch-name` is the name you want to give your local branch
 * Do your work.
-  - Make changes to the `.yml` files under the `schemas` directory.
+  - Schema changes will be done in the `.yml` files under the `schemas` directory.
+  - Generator scripts and other tooling resides in the `scripts` directory.
+  - Generated artifacts fall under the `generated` directory.
 * Run `make` to update generated files.
   - Note that the README.md file is generated, and should not be edited directly. Source files are in the /docs directory.
+* If necessary, make sure tests pass.
+  - `make check`
```

Comment:
For running tests, I rather recommend `make test`, which strictly runs the Python tests.

`make check` is an omnibus task that runs the tests, but also checks the code formatting and changes the code in place, and finally it fails if there's anything not checked in (aka if code formatting changed any files). So it's not the best when running the tests while working on a PR